core/src/main/java/tc/oc/pgm/community/modules/FreezeMatchModule.java

```diff
@@ -0,0 +1,349 @@
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.entity.TNTPrimed;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.player.PlayerBucketEmptyEvent;
+import org.bukkit.event.player.PlayerBucketFillEvent;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+import org.bukkit.event.player.PlayerDropItemEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.event.vehicle.VehicleDamageEvent;
+import org.bukkit.event.vehicle.VehicleEnterEvent;
+import org.bukkit.event.vehicle.VehicleExitEvent;
+import org.bukkit.event.vehicle.VehicleMoveEvent;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.util.Vector;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.Permissions;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.event.ObserverInteractEvent;
+import tc.oc.pgm.community.commands.ModerationCommands;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.listeners.ChatDispatcher;
+import tc.oc.pgm.spawns.events.ObserverKitApplyEvent;
+import tc.oc.util.bukkit.OnlinePlayerMapAdapter;
+import tc.oc.util.bukkit.chat.Sound;
+import tc.oc.util.bukkit.component.Component;
+import tc.oc.util.bukkit.component.Components;
+import tc.oc.util.bukkit.component.types.PersonalizedTranslatable;
+import tc.oc.util.bukkit.named.NameStyle;
+import tc.oc.util.bukkit.translations.AllTranslations;
+
+@ListenerScope(MatchScope.LOADED)
+public class FreezeMatchModule implements MatchModule, Listener {
+
+  private static final Material TOOL_MATERIAL = Material.ICE;
+  private static final int TOOL_SLOT_NUM = 6;
+
+  private final Match match;
+  private final Freeze freeze;
+
+  public FreezeMatchModule(Match match) {
+    this.match = match;
+    this.freeze = new Freeze();
+  }
+
+  public Freeze getFreeze() {
+    return freeze;
+  }
+
+  public List<MatchPlayer> getFrozenPlayers() {
+    return freeze.frozenPlayers.values().stream().collect(Collectors.toList());
+  }
+
+  private ItemStack getFreezeTool(CommandSender viewer) {
+    ItemStack stack = new ItemStack(TOOL_MATERIAL);
+    ItemMeta meta = stack.getItemMeta();
+    meta.setDisplayName(
+        ChatColor.DARK_AQUA
```

Comment:
```suggestion
        ChatColor.AQUA
```

Reply:
Hey @Pablete1234,
I want to keep all the tool names as separate colors, the observer tool is already Aqua. Do you have any recommendations? Perhaps bolded white or would you rather I swap the colors between the observer tools and this?

Reply:
I’m marking this as resolved, please check my most recent comment below for the new format. I think it looks better than dark aqua. Let me know if you agree.